src/DwitTech.AccountService.Core/Services/UserService.cs

```diff
@@ -153,16 +153,7 @@ public async Task<bool> ChangePasswordAsync(string currentPassword, string newPa
                 throw new ArgumentException("Invalid current password");
             }
 
-            if (currentPasswordHash == newPasswordHash)
```

Comment:
I think this condition should be left here, as it is still relevant to this method, and the check is not done in any of the new methods you implemented.